apps/arweave/src/ar_nonce_limiter.erl

```diff
@@ -1189,17 +1187,13 @@ maybe_set_vdf_metrics(SessionKey, CurrentSessionKey, Session) ->
 			ok
 	end.
 
-trigger_computed_outputs(_SessionKey, _Session, _PrevSessionKey, _PrevSession,
-		_UpperBound, []) ->
+send_events_for_external_update(_SessionKey, _Header, _UpperBound, []) ->
 	ok;
-trigger_computed_outputs(SessionKey, Session, PrevSessionKey, PrevSession, UpperBound,
-		[Step | Steps]) ->
-	ar_events:send(nonce_limiter, {computed_output, {SessionKey, Session, PrevSessionKey,
-			PrevSession, Step, UpperBound}}),
+send_events_for_external_update(SessionKey, Session, UpperBound, [Step | Steps]) ->
 	#vdf_session{ step_number = StepNumber, steps = [_ | PrevSteps] } = Session,
-	Session2 = Session#vdf_session{ step_number = StepNumber - 1, steps = PrevSteps },
-	trigger_computed_outputs(SessionKey, Session2, PrevSessionKey, PrevSession, UpperBound,
-			Steps).
+	ar_events:send(nonce_limiter, {computed_output, {SessionKey, StepNumber, Step, UpperBound}}),
```

Comment:
We do not seem to need the manipulations with Session#vdf_session.steps anymore, we only need to track the StepNumber, which might make sense to pass as another function argument.

Reply:
Ooh, good call. I love it! More code deletion!

Reply:
Fixed: https://github.com/ArweaveTeam/arweave/pull/498/commits/9818fdfdf7e4a1bb2c87b83cd1f86d82bb1e58d5